text/0000-declarative-root.md

```diff
@@ -0,0 +1,276 @@
+
+```html
+<template>
+    <lwc:root class="foo"></lwc:root> <!-- Doesn't actually render -->
+</template>
+```
+
+However, many attributes and directives that can be applied to a normal `HTMLElement` can be applied to `<lwc:root>`. These include:
+
+- Standard [global HTML attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes), such as `class` and `tabindex`
+- Custom HTML attributes, such as `data-*`
+- Event listeners, such as `onclick` or `onfocus`
+- The `lwc:spread` directive
+
+Other directives that apply to normal `HTMLElement`s, such as `lwc:inner-html`, `lwc:ref`, and `key`, are not supported.
+
+A bare `<lwc:root></lwc:root>` with no attributes/listeners/directives is allowed, but effectively does nothing.
+
+### Restrictions
+
+#### Placement
+
+A `<lwc:root>` element must be placed at the top level (root) of a `<template>`, and may not be preceded by other elements:
+
+```html
+<!-- Valid -->
+<template>
+    <lwc:root></lwc:root>
+    <div></div>
+</template>
+```
+
+```html
+<!-- Invalid -->
+<template>
+    <div>
+        <lwc:root></lwc:root>
+    </div>
+</template>
+```
+
+```html
+<!-- Invalid -->
+<template>
+    <div></div>
+    <lwc:root></lwc:root>
+</template>
+```
+
+Comments are allowed to precede `<lwc:root>`:
+
+```html
+<!-- Valid -->
+<template>
+    <!-- Comment -->
+    <lwc:root></lwc:root>
+</template>
+```
+
+However, in the case of `lwc:preserve-comments`, comments may not precede `<lwc:root>`:
+
+```html
+<!-- Invalid -->
+<template lwc:preserve-comments>
+    <!-- Comment -->
+    <lwc:root></lwc:root>
+</template>
+```
+
+#### Contents
+
+The `<lwc:root>` element may not have any contents other than whitespace and comments:
+
+```html
+<!-- Valid -->
+<template>
+    <lwc:root>
+        
+    </lwc:root>
+</template>
+```
+
+```html
+<!-- Valid -->
+<template>
+    <lwc:root>
+        <!-- Comment -->
+    </lwc:root>
+</template>
+```
+
+However, in the case of `lwc:preserve-comments`, comments inside of `<lwc:root>` are not allowed:
+
+```html
+<!-- Invalid -->
+<template lwc:preserve-comments>
+    <lwc:root>
+        <!-- Comment -->
+    </lwc:root>
+</template>
+```
+
+This restriction is why `lwc:inner-html` is not supported on `<lwc:root>`.
+
+### Timing
+
+In terms of timing, any attributes added to or removed from the root element should follow the timing applied
+to elements inside of the template:
+
+```html
+<template>
+    <lwc:root class={clazz}></lwc:root>
+    <div class={clazz}></div>
+</template>
+```
+
+In other words, in the above example, if `clazz` changes, then the root element's `class` attribute should be changed
+in the same tick as when the `<div>`'s `class` changes.
+
+The above also applies to the timing of when event listeners are attached using `on*`.
+
+No guarantees are made about the ordering of changes made to `<lwc:root>` versus elements inside of the template.

```

Comment:
> ordering of changes made to `<lwc:root>`

Is this referring to the ordering of how classes, event handlers, etc. are applied from `<lwc:root>` to the host element?

> versus elements inside of the template

Does this mean that there could be a difference in the order the classes, event handlers, etc are applied on the host vs to elements within the host?

Reply:
No, I didn't think of that. Updated! bbbc288